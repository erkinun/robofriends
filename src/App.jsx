import React from 'react'
import { useSelector, useDispatch } from 'react-redux'
import CardList from './components/CardList'
import SearchBox from './components/SearchBox'
import Scroll from './components/Scroll'
import ErrorBoundary from './components/ErrorBoundary'
import { setSearchField } from './store/search'
import Header from './components/Header'

const App = () => {
  const robots = useSelector((state) => state.fetchRobotsReducer.robots)
  const isPending = useSelector((state) => state.fetchRobotsReducer.isPending)
  const searchField = useSelector((state) => state.searchRobotsReducer.searchField)
  const dispatch = useDispatch()

  const filtered = robots.filter((r) => r.name.toLowerCase().includes(searchField))

  // TODO might want to fix the two states by just saving the search field state
  return (
    <div className='tc'>
      <Header />
      <SearchBox onSearch={(e) => dispatch(setSearchField(e.target.value))} searchField={searchField} />
      <Scroll>
        <ErrorBoundary>
          {isPending && <h2>Robots are coming!</h2>}
          <CardList robots={filtered} />
        </ErrorBoundary>
      </Scroll>
    </div>
  )
}

export default App
