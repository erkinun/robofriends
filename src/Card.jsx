import React from 'react'

const card = ({ name }) => {
  const url = `https://robohash.org/${name}`
  return (
    <div>
      <img alt='photo' src={url} />
      <div>
        <h2>{name}</h2>
        <p>jane.doe@gmail.com</p>
      </div>
    </div>
  )
}

export default card
