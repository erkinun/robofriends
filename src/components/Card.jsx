import React from 'react'

const card = ({ id, name, username, email }) => {
  const url = `https://robohash.org/${id}`
  return (
    <div className='bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5'>
      <img alt='robot' src={url} />
      <div>
        <h2>{name}</h2>
        <p>{email}</p>
      </div>
    </div>
  )
}

export default card
