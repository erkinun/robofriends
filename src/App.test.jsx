import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'
import { createStore } from 'redux'

import App from './App'
import Card from './components/Card'
import reducer from './store/reducers'

describe('<App /> unit test or integration test?', () => {
  const mockStore = createStore(reducer, {
    fetchRobotsReducer: {
      robots: [
        {
          id: 1,
          name: 'Leanne Graham',
          username: 'Bret',
          email: 'Sincere@april.biz',
        },
        {
          id: 2,
          name: 'Ervin Howell',
          username: 'Antonette',
          email: 'Shanna@melissa.tv',
        },
        {
          id: 3,
          name: 'Clementine Bauch',
          username: 'Samantha',
          email: 'Nathan@yesenia.net',
        },
      ],
    },
  })
  const getWrapper = () =>
    mount(
      <Provider store={mockStore}>
        <App />
      </Provider>
    )
  it('renders application', () => {
    expect(getWrapper()).toMatchSnapshot()
  })

  it('renders some robots', () => {
    expect(getWrapper().find(Card).length).toBe(3)
  })
})
