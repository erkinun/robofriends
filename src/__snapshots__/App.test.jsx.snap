// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> unit test or integration test? renders application 1`] = `ReactWrapper {}`;
